File: src/features/typing/data/go/game_of_life.go
package main

func gameOfLife(board [][]int) {
	dirs := [8][2]int{{-1, -1}, {-1, 0}, {-1, 1}, {0, -1}, {0, 1}, {1, -1}, {1, 0}, {1, 1}}
	for i, row := range board {
		for j := range row {
			live := 0
			for _, d := range dirs {
				x, y := i+d[0], j+d[1]
				if x >= 0 && x < len(board) && y >= 0 && y < len(row) && board[x][y]%2 == 1 {
					live++
				}
			}
			if (board[i][j] == 1 && (live == 2 || live == 3)) || (board[i][j] == 0 && live == 3) {
				board[i][j] += 2
			}
		}
	}
	for i, row := range board {
		for j := range row {
			board[i][j] %= 2
		}
	}
}

File: src/features/typing/data/dart/game_of_life.dart
class Solution {
  void gameOfLife(List<List<int>> board) {
    const dirs = [
      [-1, -1],
      [-1, 0],
      [-1, 1],
      [0, -1],
      [0, 1],
      [1, -1],
      [1, 0],
      [1, 1],
    ];
    for (int i = 0; i < board.length; i++) {
      for (int j = 0; j < board[i].length; j++) {
        int live = 0;
        for (final d in dirs) {
          final int x = i + d[0];
          final int y = j + d[1];
          if (x >= 0 &&
              x < board.length &&
              y >= 0 &&
              y < board[i].length &&
              board[x][y] % 2 == 1) {
            live++;
          }
        }
        if ((board[i][j] == 1 && (live == 2 || live == 3)) ||
            (board[i][j] == 0 && live == 3)) {
          board[i][j] += 2;
        }
      }
    }
    for (int i = 0; i < board.length; i++) {
      for (int j = 0; j < board[i].length; j++) {
        board[i][j] %= 2;
      }
    }
  }
}
